### PR TITLE
ci: add codecov.yml

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Add codecov workflow

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: .github/workflows/codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  codecov:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "^1.19"
+      - name: Run tests
+        run: make test
+      - uses: codecov/codecov-action@v3
```
Comment (sozercan) [inline .github/workflows/codecov.yaml:20]: i would like to pin to digests and let dependabot update these

File: .github/workflows/codecov.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Codecov
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
```
Comment (sozercan) [inline .github/workflows/codecov.yaml:9]: permissions: read-all

**Review by sozercan — APPROVED** [review]

File: .github/workflows/codecov.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+permissions: read-all
```
Comment (enj) [inline .github/workflows/codecov.yaml:10]: There aren't any repo tokens that a user running from a fork could get right?

Comment (aramase) [reply .github/workflows/codecov.yaml:10]: No, there aren't!

**Pull request merged**